test/runtests.jl

```diff
@@ -490,4 +490,25 @@ end
     end
 end
 
+@testset "_ndownload_tasks" begin
+    MB = 1024*1024
+    @test CloudStore.API._ndownload_tasks(32MB, 64MB, 8) == 1
+    @test CloudStore.API._ndownload_tasks(32MB, 32MB, 8) == 1
+    @test CloudStore.API._ndownload_tasks(32MB, 16MB, 8) == 2
+    @test CloudStore.API._ndownload_tasks(32MB, 8MB, 8) == 4
+    @test CloudStore.API._ndownload_tasks(32MB, 4MB, 8) == 8
+    @test CloudStore.API._ndownload_tasks(32MB, 2MB, 8) == 8
+
+    @test CloudStore.API._ndownload_tasks(32MB, 16MB + 1, 8) == 2
+    @test CloudStore.API._ndownload_tasks(32MB, 16MB - 1, 8) == 3
+    @test CloudStore.API._ndownload_tasks(32MB, 8MB + 1, 8) == 4
+    @test CloudStore.API._ndownload_tasks(32MB, 8MB - 1, 8) == 5
+    @test CloudStore.API._ndownload_tasks(32MB, 4MB + 1, 8) == 8
+    @test CloudStore.API._ndownload_tasks(32MB, 4MB - 1, 8) == 8
+
+    @test CloudStore.API._ndownload_tasks(32MB, 1, 8) == 8
+    @test CloudStore.API._ndownload_tasks(32MB, 32MB, 8) == 1
+    @test CloudStore.API._ndownload_tasks(32MB, 64MB, 8) == 1
+end
```

Comment:
This is great, thanks!

Is the size of the file allowed to be 0? If so, it'd be nice to have an example for that.
Also, some tests for when you only have 1 thread.

Thanks :)

Reply:
Yes, cloud objects can have 0 size.

Reply:
But we wouldn't spawn any tasks in that case (we wouldn't even hit `_ndownload_tasks`)

Reply:
Makes sense. Then instead, we could have an assertion in that function that size != 0.

Alternatively, it's totally okay for this function to have a wider acceptable set of inputs than we actually use! That's _why_ it's important to _unit test_ the entire range of an API's accepted input set, since you might not be exercising it through integration tests since the system isn't using it, but **the way it's used might change in the future,** and you want to be sure that the functionality stays working.

So basically, for a public API in a package, i think it's important to identify what the set of accepted inputs are (the "[preconditions](https://en.wikipedia.org/wiki/Precondition)"), document them clearly ("size must be >= 0"), and then make sure that the function is unit tested over all the corner cases of the valid preconditions.
Optionally, you can then also add assertions that the input meets the preconditions to help out users who use it wrong, but this part is less clear cut and i think is usually decided on a case-by-case basis.

Does that make sense?

Reply:
Ah, but rereading the PR i had forgotten that this is an internal function, not a public one. So i do think the above can be relaxed quite a bit.

But still, I would recommend to do either one of the following:
1. Add a quick comment above the function "size should be >= 1", or
2. Add a unit test for size = 0, even though we aren't currently using it, just to make sure that it's tested in case we _do_ change the code to use it in the future.